common/service/task/state/state.go

```diff
@@ -141,3 +153,13 @@ func New(subStatus SubStatus, taskID string) State {
 		store:  store,
 	}
 }
+
+func isValid(statusLog map[string]interface{}) bool {
+	if statusLog != nil {
+		if len(statusLog) != 0 {
```

Comment:
I Think isValid should only check nil map and just to check nil map.. do we really need a function? 
I think we can improve the logic around here
I would also think about  defining our own type like type Fields map[string]interface{} and then writing function for it. 

func (f Fields) IsValid() bool

if not, then please consider `return len(statusLog) != 0`